python/pyarrow/tests/test_pandas.py

```diff
@@ -4996,3 +4997,7 @@ def roundtrip(df, schema=None):
     schema = pa.schema([("maps", map_type)])
     roundtrip(pd.DataFrame({"maps": [map_of_los, map_of_los, map_of_los]}),
               schema=schema)
+
+def test_is_data_frame_race_condition():
+    # See https://github.com/apache/arrow/issues/39313
+    test_util.invoke_script('arrow_39313.py')
```

Comment:
Note that the time window of this race condition starts with the first call of `_check_import` within a python process, and ends when that first call returns. Concurrent calls to `_check_import` in this time window can cause the race condition. 

Give arbitrary pytest/conftest code could call `_check_import`, we use `invoke_script` to start a new subprocess, where we can be sure our test (which reproduces the issue) call `_check_import` for the first time in that process.